package-res/resources/web/pentaho/config/util.js

```diff
@@ -0,0 +1,222 @@
+/*!
+ * Copyright 2010 - 2016 Pentaho Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define([
+  "../util/object",
+  "../lang/OperationInvalidError"
+], function(O, OperationInvalidError) {
+
+  "use strict";
+
+  /**
+   * Map of merge operation name to operation handler function.
+   *
+   * @type {Object.<string, function>}
+   * @see mergeSpecsOne
+   */
+  var _mergeHandlers = {
+    "replace": mergeSpecsOperReplace,
+    "merge": mergeSpecsOperMerge,
+    "add": mergeSpecsOperAdd
+  };
+
+  /**
+   * @classDesc A singleton class that contains utilities related to configuration.
+   *
+   * @name pentaho.config.Util
+   * @class
```

Comment:
I think this could easily belong in `pentaho.util`. A potential receiver could be `object.js`, but I would probably prefer it to live by its own (we just need a good name). Other methods could be publicly exposed, like `cloneOwnDeep`.

Lets keep this here for now, but take a note that this should be refactored out either when similar functionality is needed elsewhere or when doing an overall revision for producing separate bundles.